### PR TITLE
feat: sp1-sdk, remote prover

### PR DESCRIPTION
**Pull request by ctian1**

### CODE REVIEW

File: sdk/src/client.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    },
+    SP1ProofWithIO,
+};
+
+pub struct SP1ProverServiceClient {
+    pub rpc: TwirpClient,
+    pub http: HttpClientWithMiddleware,
+}
+
+const DEFAULT_SP1_SERVICE_URL: &str = "https://rpc.succinct.xyz/";
```
Comment (jtguibas) [inline sdk/src/client.rs:27]: PROVER_NETWORK_RPC

File: sdk/src/client.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+impl SP1ProverServiceClient {
+    pub fn with_token(access_token: String) -> Self {
+        let rpc_url =
+            env::var("SP1_SERVICE_URL").unwrap_or_else(|_| DEFAULT_SP1_SERVICE_URL.to_string());
```
Comment (jtguibas) [inline sdk/src/client.rs:32]: same here

File: sdk/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+tokio = { version = "1.36.0", features = ["full"] }
+p3-matrix = { workspace = true }
+p3-commit = { workspace = true }
+p3-field = { workspace = true }
+indicatif = "0.17.8"
+tracing = "0.1.40"
+hex = "0.4.3"
+
+[build-dependencies]
+vergen = { version = "8", default-features = false, features = ["build", "git", "git2"] }
```
Comment (jtguibas) [inline sdk/Cargo.toml:30]: it doesn't look like you use vergen?

**Review by jtguibas — APPROVED** [review]

File: sdk/service.proto [hunk 4]
```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+package prover;
+
+service SP1ProverService {
+  rpc CreateProof(CreateProofRequest) returns (CreateProofResponse) {}
+  rpc SubmitProof(SubmitProofRequest) returns (SubmitProofResponse) {}
+  rpc GetProofStatus(GetProofStatusRequest) returns (GetProofStatusResponse) {}
+}
+
```
Comment (mattstam) [inline sdk/service.proto:10]: Documenting each of these states (what they mean, how they transition) would be good

**Review by mattstam — APPROVED** [review]

**Pull request merged**